### PR TITLE
General cleanup of sweep infrastructure. Minor refactoring of AngleAggregation class.

### PR DESCRIPTION
**Pull request by wdhawkins**
This PR contains the following improvements:

1. General cleanup in `/modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep`, especially in `sweep_namespace.h`. Renaming `sweep_namespace.h` to `sweep.h`. Deleting `spls` and moving that functionality to `sweep.h`.
2. Minor refactoring of AngleAggregation class, including: proper naming of class member variables, creating a proper initializing constructor, formatting, etc.

### CODE REVIEW

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h [hunk 1]
```diff
@@ -26,38 +26,43 @@ namespace lbs
...
+                   std::shared_ptr<AngularQuadrature>& quadrature,
+                   std::shared_ptr<MeshContinuum>& grid);
 
-  /** Sets up the angle-aggregation object. */
-  AngleAggregation(const std::map<uint64_t, std::shared_ptr<SweepBndry>>& in_sim_boundaries,
-                   size_t in_number_of_groups,
-                   size_t in_number_of_group_subsets,
-                   std::shared_ptr<AngularQuadrature>& in_quadrature,
-                   std::shared_ptr<MeshContinuum>& in_grid);
+  std::vector<AngleSetGroup> angle_set_groups_;
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h:49]: I think we use trailing `_` for protected and private member variables. The public ones do not have it. See [coding standard](https://open-sn.github.io/opensn/coding_standard.html#classes-and-structs).

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h [hunk 2]
```diff
@@ -19,10 +16,18 @@ enum class FaceOrientation : short
...
+  NOT_FINISHED = 0,
+  FINISHED = 1,
+  RECEIVING = 2,
+  READY_TO_EXECUTE = 3,
+  EXECUTE = 4,
+  NO_EXEC_IF_READY = 5,
+  MESSAGES_SENT = 6,
+  MESSAGES_PENDING = 7
+};
+typedef AngleSetStatus ExecutionPermission;
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:30]: I know this is just a moved piece of code, but interesting. I would not see right away why `Status` maps to `ExecutionPermission`. I am not requesting any action ATM, just pointing that out...

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h [hunk 3]
```diff
@@ -26,38 +26,43 @@ namespace lbs
  * The octant based separation is achieved via the notion of "AngleSetGroup"
  * which will group angle sets for each quadrant or octant
  * (depending on 2D or 3D).*/
+
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h:29]: Could you move the closing `*/` on its own line and have the doxygen comment attached to the class declaration, i.e. no empty line like this?

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h [hunk 4]
```diff
@@ -26,38 +26,43 @@ namespace lbs
...
-
 private:
-  bool is_setup = false;
-  std::pair<size_t, size_t> number_angular_unknowns;
-  bool num_ang_unknowns_avail = false;
+  bool is_setup_;
+  size_t number_of_groups_;
+  size_t number_of_group_subsets_;
+  bool num_ang_unknowns_avail_;
+  std::pair<size_t, size_t> number_angular_unknowns_;
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h:37]: Could you unify this so that the variable names are `num_` instead of `number_`? 
From the code `num_` vs. `number_` is ~4000 vs. ~100. 
This could be a separate patch.

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h [hunk 5]
```diff
@@ -33,14 +38,18 @@ struct Task
...
 };
 
-class AAH_ASynchronousCommunicator;
-class AngleSet;
-class AngleSetGroup;
-class AngleAggregation;
-
-class SweepChunk;
+/**Contains the sweep plane data.*/
+struct SPLS
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:42]: Just a question: Should this be SPLD as in `S`weep `PL`ane `D`ata or does the trailing `S` stands for something else?

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h (hunk 1 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h:49]: Oh, good catch! I originally made that private and didn't fix it when I made it public.

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h (hunk 2 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:30]: Me either, and I've removed this in a future commit.

Comment (andrsd) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:30]: Ok, good :+1:

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h (hunk 5 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:42]: SPLS -> Sweep Plane Local Subgrid

Comment (andrsd) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep.h:42]: That seems like that should be in the comment then?

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h (hunk 1 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h]: Fixed.

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h (hunk 4 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h:37]: Fixed.

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h (hunk 3 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.h]: Fixed.

**Review by andrsd — APPROVED** [review]

**Review by ragusa — APPROVED** [review]

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.cc [hunk 6]
```diff
@@ -10,22 +10,23 @@ namespace lbs
 {
 
 AngleAggregation::AngleAggregation(
-  const std::map<uint64_t, std::shared_ptr<SweepBndry>>& in_sim_boundaries,
-  size_t in_number_of_groups,
-  size_t in_number_of_group_subsets,
-  std::shared_ptr<AngularQuadrature>& in_quadrature,
-  std::shared_ptr<MeshContinuum>& in_grid)
+  const std::map<uint64_t, std::shared_ptr<SweepBndry>>& sim_boundaries,
```
Comment (zhardy-lanl) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.cc:13]: I know this is not a change made in this PR, but it would be nice to just spell out `SweepBoundary`. Additionally, any idea what a `sim_boundary` is versus a `boundary`? Feel free to punt on this.

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep_scheduler/sweep_scheduler.cc [hunk 7]
```diff
@@ -238,7 +238,7 @@ SweepScheduler::ScheduleAlgoDOG(SweepChunk& sweep_chunk)
     for (auto& angle_set : angle_set_group.AngleSets())
       angle_set->ResetSweepBuffers();
 
-  for (auto& [bid, bndry] : angle_agg_.sim_boundaries)
+  for (auto& [bid, bndry] : angle_agg_.GetSimBoundaries())
```
Comment (zhardy-lanl) [inline modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/sweep_scheduler/sweep_scheduler.cc:241]: Same comment on `Sim`

**Review by zhardy-lanl — APPROVED** [review]

File: modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.cc (hunk 6 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep/angle_aggregation/angle_aggregation.cc:13]: I address SweepBndry -> SweepBoundary in a future commit.  I'll also addres sim_boundaries there as well.
